framework/src/kernels/Kernel.C

```diff
@@ -80,6 +80,25 @@ Kernel::computeJacobian()
 
   ke += _local_ke;
 
+  if (_calculate_nonlocal_variable_jacobian)
+  {
+    _nonlocal_ke.resize(ke.m(), _subproblem.es().n_dofs());
+    _nonlocal_ke.zero();
+    for (_i = 0; _i < _test.size(); _i++)
```

Comment:
Wow - this is a scary looking loop!  Looks like it's going to take a crap-ton of time...

I'll be interested to see the timing on this stuff!